client/app/src/main/java/com/microsoft/research/karya/ui/registration/OTPFragment.kt

```diff
@@ -15,17 +19,88 @@ import com.microsoft.research.karya.ui.base.BaseActivity
 import com.microsoft.research.karya.utils.viewBinding
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import com.microsoft.research.karya.ui.registration.RegistrationViewModel.OtpVerifyState
+import com.microsoft.research.karya.ui.registration.RegistrationViewModel.OtpSendState
+import dagger.hilt.android.AndroidEntryPoint
 
 private const val OTP_LENGTH = 6
 
+@AndroidEntryPoint
 class OTPFragment : Fragment(R.layout.fragment_o_t_p) {
 
     private val binding by viewBinding(FragmentOTPBinding::bind)
+    private val viewModel by viewModels<RegistrationViewModel>()
```

Comment:
```suggestion
    private val viewModel by activityViewModels<RegistrationViewModel>()
```
This is the registration flow so we want the viewModel to be shared among all fragments.

Reply:
I was initialising the same viewmodel using viewModels. I thought it would use the same instance of viewmodel along the fragements. I will update this